File: Arrays/2574. Left and Right Sum Differences.py
from functools import reduce


def leftRightDifference(nums):
    answer = []
    for i in range(len(nums)):
        leftSum = sum(nums[0:i])
        rightSum = sum(nums[i+1:len(nums)])
        answer.append(abs(leftSum-rightSum))
    return answer

print(leftRightDifference([10,4,8,3]))

File: Arrays/2574. Left and Right Sum Differences.js
link = 'https://leetcode.com/problems/left-and-right-sum-differences/'


// Solution
//---------------------------------
const leftRightDifference = (nums) => {
    let answer = []
    for (let i = 0; i < nums.length; i++) {
        let leftsum = nums.slice(0,i).reduce((carry, element)=>{
            return carry+element
        },0)
        let rightsum = nums.slice(i+1,nums.length).reduce((carry, element)=>{
            return carry+element
        },0)
        answer.push(Math.abs(leftsum-rightsum))
    }
    return answer
}
// //---------------------------------
let result = leftRightDifference([10,4,8,3])
console.log(result)

